pilot/pkg/model/service.go

```diff
@@ -769,6 +780,20 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 	return s.DefaultAddress
 }
 
+// getAllAddresses returns a Service's all addresses.
+func (s *Service) getAllAddresses() []string {
+	var addresses []string
+	addressMap := s.ClusterVIPs.GetAddresses()
+	for _, clusterAddresses := range addressMap {
+		addresses = append(addresses, clusterAddresses...)
+	}
+
+	if len(addresses) == 0 && s.DefaultAddress != "" {
```

Comment:
DefaultAddress  can be "0.0.0.0"? 
In this case, there maybe autoallocated address, do we need to return it?

Reply:
> DefaultAddress can be "0.0.0.0"? In this case, there maybe autoallocated address, do we need to return it?

We shouldn't return “0.0.0.0" or the auto-allocated address.